common/app/services/newsletters/NewsletterApi.scala

```diff
@@ -0,0 +1,95 @@
+package services.newsletters
+
+import com.gu.identity.model.{EmailEmbed, NewsletterIllustration}
+import common.{BadConfigurationException, Logging}
+import conf.Configuration._
+import play.api.libs.json.{JsResult, JsValue, Json}
+import play.api.libs.ws.WSClient
+
+import scala.concurrent.duration.DurationInt
+import scala.concurrent.{ExecutionContext, Future}
+
+case class NewsletterResponse(
+    id: String,
+    name: String,
+    brazeNewsletterName: String,
+    brazeSubscribeAttributeName: String,
+    brazeSubscribeEventNamePrefix: String,
+    theme: String,
+    description: String,
+    frequency: String,
+    exactTargetListId: Int,
+    emailEmbed: EmailEmbed,
+    illustration: Option[NewsletterIllustration] = None,
+)
+
+case class GroupedNewsletterResponse(
+    displayName: String,
+    newsletters: List[NewsletterResponse],
+)
+
+case class GroupedNewslettersResponse(
+    newsRoundups: GroupedNewsletterResponse,
+    newsByTopic: GroupedNewsletterResponse,
+    features: GroupedNewsletterResponse,
+    sport: GroupedNewsletterResponse,
+    culture: GroupedNewsletterResponse,
+    lifestyle: GroupedNewsletterResponse,
+    comment: GroupedNewsletterResponse,
+    work: GroupedNewsletterResponse,
+    fromThePapers: GroupedNewsletterResponse,
+)
+
+case class NewsletterApi(wsClient: WSClient)(implicit executionContext: ExecutionContext)
+    extends Logging
+    with implicits.WSRequests {
+
+  implicit val emailEmbedReads = Json.reads[EmailEmbed]
+
+  implicit val newsletterIllustrationReads = Json.reads[NewsletterIllustration]
+
+  implicit val newsletterResponseReads = Json.reads[NewsletterResponse]
+
+  implicit val groupedNewsletterResponseReads = Json.reads[GroupedNewsletterResponse]
+
+  implicit val groupedNewslettersReads = Json.reads[GroupedNewslettersResponse]
```

Comment:
Ideally these should be moved to a companion object (the static part of the code). In this case it doesn't matter _that_ much, be we tend to put parsers on companion objects as you really only need one particular instance of a parser for your whole application. 

Otherwise if we were to create two instances of `NewsletterApi`, we'd have as many parsers in memory as we have instances of the `NewsletterApi`. It would be fine for `NewsletterApi`, but let's imagine these were on a class called `User` or `Request` and suddenly we're creating lots of parsers for no good reasons.

Reply:
I tried moving this to a companion object just now and got a bunch of compilations errors where the class methods couldn't find the `implicit`s. I imagine it's just me implementing incorrectly and is something to handle pairing on it